docs/guides/blocks/sync-tables.md

```diff
@@ -2,9 +2,224 @@
+
+The item is displayed as a chip in the first column of the sync table, and it contains the complete set of synced data. Properties of the schema designated as [`featured`][featured] are automatically broken out into their own columns, but these can be removed or added to by the user later.
+
+The schema used in a sync table can be shared by other building blocks in your Pack, like a formula or column format. Even when the schema is only used by the sync table, we recommend you define it as a separate variable in your code for readability and future reuse.
+
+```ts
+const TaskSchema = coda.makeObjectSchema({
+  // ...
+});
+```
+
+!!! info "Dynamic schemas"
+    For some data sources it isn't possible to know the shape of the data upfront, and it can only be determined at runtime. In these cases instead of defining your schema upfront you can generate it dynamically. See the [Dynamic sync tables][dynamic_sync_tables] guide for more information.
+
+
+### Define the sync table
+
+The sync table itself is defined using the `addSyncTable()` method:
+
+```ts
+pack.addSyncTable({
+  name: "Tasks",
+  schema: TaskSchema,
+  identityName: "Task",
+  formula: {
+    // ...
+  },
+});
+```
+
+It includes the name of the sync table, the schema for each row, and other metadata. The `identityName` property will be explained in the [Identity](#identity) section below.
+
+
+### Write the sync formula
+
+Inside each sync table definition is a formula definition, detailing the hidden formula used to sync data into the sync table. If you aren't already familiar with creating formulas, read the [Formulas guide][formulas] first.
+
+```ts
+pack.addSyncTable({
+  // ...
+  formula: {
+    name: "SyncTasks",
+    description: "Sync the user's tasks.",
+    parameters: [],
+    execute: async function ([], context) {
+      // ...
+    }
+  },
+});
+```
+
+This formula fetches items from the external data source, adjust them to fit the schema, and returns them. The return value must be a [`SyncFormulaResult`][SyncFormulaResult], which is an object where `result` property contains the array of item objects.
+
+```ts
+pack.addSyncTable({
+  // ...
+  formula: {
+    // ...
+    execute: async function ([], context) {
+      let url = "<URL to pull data from>";
+      let response = await context.fetcher.fetch({
+        method: "GET",
+        url: url,
+      });
+      let items = response.body.items;
+      // Adjust the items to fit the schema if required.
+      return {
+        result: items,
+      }
+    }
+  },
+});
+```
+
+Your formula must return the full set of items on each sync, and Coda will determine how to update existing rows, remove old rows, etc. This could take a long time for large datasets or slow APIs, and the [Long-running syncs](#longrunning) section has more information on how to handle this.
```

Comment:
I know this is trying to bite off only one concept at a time, but I think "your formula must return the full set of items on each sync" is bordering on inaccurate. A sync invocation consisting of repeated formula invocations must return the full set of items, but if you didn't already know about pagination this copy would strongly suggest that one formula invocation should return the whole result set and if you don't read the other section you would never be disabused of that notion.

I think this paragraph should address some of the nuance and make clear that a single formula invocation should only ever return a reasonable-sized page of results, and link out for the longer explanation.

Reply:
That a good callout. There is _so much_ detail to learn, I try to trickle it in as slow as I can 😄. What about his updated language, "On each sync..."? It's factually accurate without front-loading any additional complexity about pagination.